Better OSM explanation in login screen

map-per opened the pull request:
Added a better OSM explanation to the login screen. The explanations is most important for users that never heard about OSM before. These users won't select the login screen in the settings, but see the screen directly after editing or adding a place. We need a good explanation to make sure as many users as possible log in to share their edits on OSM.

This PR adds an explanation about what OSM is and why users should create a new account / need one  for uploading.

Also fixed margins in the google flavor.

### Roadmap

- [x] Settle on the UI design and the english strings
- [x] Update UI in horizontal screen
- [x] Localize strings in `strings.txt`
- [x] Translation review

### Google

| before | now |
|--------|--------|
| ![Screenshot_2024-09-28-12-39-43-203_app organicmaps](https://github.com/user-attachments/assets/3455aabe-ea39-42f7-83af-fd3b4ba6726e) | ![Screenshot_2024-10-25-17-16-28-753_app organicmaps debug](https://github.com/user-attachments/assets/d2412304-b491-4fa4-9ae3-6bf5a69cdd03) | 

|  |  |
|--------|--------|
| before | ![Screenshot_2024-10-19-19-13-37-665_app organicmaps](https://github.com/user-attachments/assets/aa9a1215-47e2-4f76-b39e-bd4e0bb23af8) |
| now | ![Screenshot_2024-10-25-17-16-34-197_app organicmaps debug](https://github.com/user-attachments/assets/b375c414-50e7-482e-8618-2046ca5430e1) | 

### F-Droid, Web

| before | now |
|--------|--------|
| ![Screenshot_2024-09-28-12-56-01-071_app organicmaps debug](https://github.com/user-attachments/assets/51df2e15-8b24-4291-9c87-ffc6e484bd11) | ![Screenshot_2024-10-25-16-21-03-025_app organicmaps debug](https://github.com/user-attachments/assets/1b7190e8-2323-4c62-bb95-870ad399e8a6) | 

|  |  |
|--------|--------|
| before | ![Screenshot_2024-10-19-19-13-17-644_app organicmaps beta](https://github.com/user-attachments/assets/3c935b34-5f5f-4874-a60c-b643e132ef42) |
| now | ![Screenshot_2024-10-25-16-21-08-620_app organicmaps debug](https://github.com/user-attachments/assets/25ca11b9-0b29-43ca-aff1-d631ffe2433d) |

Jean-BaptisteC commented:
IMO, text is too long, it's takes too much space on the screen.

map-per commented on android/app/src/main/res/layout/fragment_osm_login.xml:
```diff
@@ -43,29 +53,19 @@
           android:fontFamily="@string/robotoRegular"
-          android:text="@string/osm_presentation"
-          android:textAppearance="@style/MwmTextAppearance.Body2"
+          android:text="OpenStreetMap.org (OSM) is a community project to build a free and open map. It's like Wikipedia for maps, you can add or edit places and they become available to millions of users all over the World.\nAs all map data in Organic Maps comes form OSM, contributing to OSM means contributing to Organic Maps. Join the community and help to make a better map for everyone!"
```
Thanks, good catch

RedAuburn commented:
I like it, if there's anywhere that should have a long text about OSM, it should be the login screen 👍

It's not like the space is needed for anything else, afaik

map-per commented:
> IMO, text is too long, it's takes too much space on the screen.

I think it's important to explain why users have to create yet another account. Account creation is a step that has a high risk of people quitting the process just because they consider creating new accounts to be annoying.

The text also bridges the gap and explains the connection between editing and OSM login / account creation.

<details><summary>I took some time to think about alternatives, but couldn't come up with good ones:</summary>
<p>

I thought about adding a new "Thanks for contributing to OSM" screen after editing and before login. The intention was to bridge the gap between editing and suddenly being prompted to log in. The idea was to inform users that their edit was saved successfully on their phone but that they need to login as the next step to publish their changes globally.

But this turned out to be a bad idea because it is confusing to distinguish between locally saved and uploaded as well as to thank the users before the process is completed. And of course adding a new screen means one more screen to read and click through for users.

</p>
</details>

map-per commented:
Inspired by https://github.com/organicmaps/organicmaps/issues/3371#issuecomment-1323481930 I also updated the landscape layout (see first post for screenshots).

map-per commented:
@Jean-BaptisteC How about this shortened version of the text:

```
OpenStreetMap.org (OSM) is a community 
project to build a free and open map. It's the 
source of all map data in Organic Maps and 
works similar to Wikipedia. You can add or 
edit places and they become available to 
millions of users all over the World.\n
Join the community and help to make a better 
map for everyone!
```

map-per commented:
Ok, I updated the string and added translations. Are we ready to call in the translation reviewers?

pastk commented:
Could you please update the screenshots with the latest text?

map-per commented:
:+1: Updated the screenshots

pastk commented on data/strings/strings.txt:
```diff
@@ -15968,9 +15968,14 @@
     zh-Hans = 截至 %@ 的社区创建的 OpenStreetMap 数据。请访问 OpenStreetMap.org 以了解有关如何编辑和更新地图的信息。
     zh-Hant = 截至 %@ 的社群創建的 OpenStreetMap 資料。請訪問 OpenStreetMap.org 以了解有關如何編輯和更新地圖的更多資訊。
 
+  [login_osm_presentation]
```
so above `[osm_presentation]` is not used in Android anymore - please remove the tag

map-per replied:
It's till used in the About OSM screen

pastk reviewed (approved):
LGTM

map-per commented:
The vast majority of data is form OSM, so saying "all" for simplicity is OK IMO.

matheusgomesms commented:
> The vast majority of data is form OSM, so saying "all" for simplicity is OK IMO.

I agree, but at the same time I was thinking of future-proofing the app (and translations, in my case).

pastk commented:
"It's the main source of map data in Organic Maps..." ?

map-per commented:
Changed "all" to "main source" and updated the translation strings.

@organicmaps/translations
Please take a look at the new _[login_osm_presentation]_ and _[login_to_make_edits_visible]_ strings. The texts are used on the login screen (see https://github.com/organicmaps/organicmaps/pull/9384#issue-2554198357 for screenshots).

biodranik commented on data/strings/strings.txt:
```diff
@@ -15968,50 +15968,98 @@
+    af = OpenStreetMap.org (OSM) is 'n gemeenskapsprojek om 'n gratis en oop kaart te bou. Dit is die hoofbron van kaartdata in Organic Maps en werk soortgelyk aan Wikipedia. Jy kan plekke byvoeg of wysig en hulle word beskikbaar vir miljoene gebruikers regoor die wêreld.\nSluit aan by die gemeenskap en help om 'n beter kaart vir almal te maak!
+    ar = OpenStreetMap.org (OSM) هو مشروع مجتمعي لبناء خريطة مجانية ومفتوحة. إنه المصدر الرئيسي لبيانات الخرائط في الخرائط العضوية ويعمل على غرار ويكيبيديا. يمكنك إضافة الأماكن أو تعديلها وتصبح متاحة لملايين المستخدمين في جميع أنحاء العالم. \انضم إلى المجتمع وساعد في إنشاء خريطة أفضل للجميع!
+    az = OpenStreetMap.org (OSM) pulsuz və açıq xəritə yaratmaq üçün icma layihəsidir. O, Üzvi Xəritələrdə xəritə məlumatlarının əsas mənbəyidir və Vikipediyaya oxşar işləyir. Siz yerlər əlavə edə və ya redaktə edə bilərsiniz və onlar bütün dünyada milyonlarla istifadəçi üçün əlçatan olur.\nİcmaya qoşulun və hər kəs üçün daha yaxşı xəritə yaratmağa kömək edin!
+    be = OpenStreetMap.org (OSM) - гэта грамадскі праект па стварэнні бясплатнай і адкрытай карты. Гэта асноўная крыніца картаграфічных даных у Organic Maps і працуе падобна на Вікіпедыю. Вы можаце дадаваць або рэдагаваць месцы, і яны стануць даступнымі для мільёнаў карыстальнікаў па ўсім свеце.\nДалучайцеся да супольнасці і дапамажыце зрабіць лепшую карту для ўсіх!
```
```suggestion
    be = OpenStreetMap.org (OSM) - гэта грамадскі праект па стварэнні бясплатнай і адкрытай мапы, і асноўная крыніца картаграфічных даных у Organic Maps, якая працуе падобна на Вікіпедыю. Кожны можа дадаваць або рэдагаваць месцы, якія пабачуць мільёны карыстальнікаў па ўсім свеце.\nДалучайцеся да супольнасці і дапамажыце зрабіць лепшую мапу для ўсіх!
```

biodranik commented on data/strings/strings.txt:
```diff
@@ -15968,50 +15968,98 @@
+    pt = O OpenStreetMap.org (OSM) é um projeto comunitário para construir um mapa livre e aberto. É a principal fonte de dados de mapas no Organic Maps e funciona de forma semelhante à Wikipedia. Pode adicionar ou editar locais e estes ficam disponíveis para milhões de utilizadores em todo o mundo.\nJunte-se à comunidade e ajude a criar um mapa melhor para todos!
+    pt-BR = O OpenStreetMap.org (OSM) é um projeto comunitário para criar um mapa gratuito e aberto. Ele é a principal fonte de dados de mapas no Organic Maps e funciona de forma semelhante à Wikipedia. Você pode adicionar ou editar lugares e eles ficam disponíveis para milhões de usuários em todo o mundo.\nJunte-se à comunidade e ajude a criar um mapa melhor para todos!
+    ro = OpenStreetMap.org (OSM) este un proiect comunitar pentru construirea unei hărți libere și deschise. Este principala sursă de date cartografice în Organic Maps și funcționează similar cu Wikipedia. Puteți adăuga sau edita locuri, iar acestea devin disponibile pentru milioane de utilizatori din întreaga lume.\nAlăturați-vă comunității și ajutați la crearea unei hărți mai bune pentru toată  ...
+    ru = OpenStreetMap.org (OSM) - это проект сообщества по созданию бесплатной и открытой карты. Он является основным источником картографических данных в Organic Maps и работает подобно Википедии. Вы можете добавлять или редактировать места, и они становятся доступными для миллионов пользователей по всему миру.\nПрисоединяйтесь к сообществу и помогите сделать лучшую карту для всех!
```
```suggestion
    ru = OpenStreetMap.org (OSM) - это открытый проект сообщества по созданию бесплатной карты. Он является основным источником картографических данных в Organic Maps и работает как Википедия. Кто угодно может добавлять или редактировать места, которые позже увидят миллионы пользователей по всему миру.\nПрисоединяйтесь к сообществу и помогите сделать лучшую карту для всех!
```

biodranik commented on data/strings/strings.txt:
```diff
@@ -15968,50 +15968,98 @@
+    af = Skep 'n OpenStreetMap-rekening of meld aan om jou kaartwysigings aan die wêreld te publiseer.
+    ar = أنشئ حساباً على OpenStreetMap أو سجّل الدخول لنشر تعديلاتك على الخريطة للعالم.
+    az = OpenStreetMap hesabı yaradın və ya xəritə redaktələrinizi dünyada dərc etmək üçün daxil olun.
+    be = Стварыце ўліковы запіс OpenStreetMap або ўвайдзіце ў сістэму, каб апублікаваць праўкі карты ўсім свеце.
```
```suggestion
    be = Стварыце ўліковы запіс OpenStreetMap або ўвайдзіце ў сістэму, каб апублікаваць праўкі мапы ўсім на свеце.
```

biodranik commented on data/strings/strings.txt:
```diff
@@ -15968,50 +15968,98 @@
+    sw = OpenStreetMap.org (OSM) ni mradi wa jumuiya wa kujenga ramani isiyolipishwa na iliyo wazi. Ndiyo chanzo kikuu cha data ya ramani katika Ramani za Kikaboni na inafanya kazi sawa na Wikipedia. Unaweza kuongeza au kubadilisha maeneo na yanapatikana kwa mamilioni ya watumiaji kote Ulimwenguni.\nJiunge na jumuiya na usaidie kutengeneza ramani bora kwa kila mtu!
+    th = OpenStreetMap.org (OSM) เป็นโครงการชุมชนเพื่อสร้างแผนที่แบบเปิดและเสรี เป็นแหล่งข้อมูลแผนที่หลักในแผนที่ทั่วไปและทำงานคล้ายกับวิกิพีเดีย คุณสามารถเพิ่มหรือแก้ไขสถานที่ได้ และสถานที่เหล่านั้นจะพร้อมให้บริการแก่ผู้ใช้หลายล้านคนทั่วโลก\nเข้าร่วมชุมชนและช่วยสร้างแผนที่ที่ดีขึ้นสำหรับทุกคน!
+    tr = OpenStreetMap.org (OSM) ücretsiz ve açık bir harita oluşturmak için bir topluluk projesidir. Organik Haritalar'daki harita verilerinin ana kaynağıdır ve Wikipedia'ya benzer şekilde çalışır. Yerleri ekleyebilir veya düzenleyebilirsiniz ve bunlar tüm dünyadaki milyonlarca kullanıcı tarafından kullanılabilir hale gelir.\nTopluluğa katılın ve herkes için daha iyi bir harita oluşturmaya yardı ...
+    uk = OpenStreetMap.org (OSM) - це громадський проект зі створення вільної та відкритої карти. Це основне джерело картографічних даних в Organic Maps і працює подібно до Вікіпедії. Ви можете додавати або редагувати місця, і вони стають доступними мільйонам користувачів по всьому світу.\nПриєднуйтесь до спільноти і допоможіть зробити карту кращою для всіх!
```
```suggestion
    uk = OpenStreetMap.org (OSM) - це громадський проект зі створення вільної та відкритої мапи. Це основне джерело картографічних даних в Organic Maps і працює подібно до Вікіпедії. Ви можете додавати або редагувати місця, і вони стають доступними мільйонам користувачів по всьому світу.\nПриєднуйтесь до спільноти і допоможіть зробити мапу кращою для всіх!
```

biodranik commented on data/strings/strings.txt:
```diff
@@ -15968,50 +15968,98 @@
+  [login_osm_presentation]
+    comment = OSM explanation on Android login screen
+    tags = android
+    en = OpenStreetMap.org (OSM) is a community project to build a free and open map. It's the main source of map data in Organic Maps and works similar to Wikipedia. You can add or edit places and they become available to millions of users all over the World.\nJoin the community and help to make a better map for everyone!
```
```suggestion
    en = OpenStreetMap.org (OSM) is an open community project to build a free map. It's the main source of map data in Organic Maps and works similarly to Wikipedia. You can add or edit places and they become available to millions of users all over the World.\nJoin the community and help to make a better map for everyone!
```

map-per replied:
I didn't mean "open" in the sense that any one can join, that's covered with "community", but "open" in the sense of open data.

pastk replied:
yeap, "free and open map" sounds better to me than just "free map" (sounds like gratis is the main property of it..)

soshial replied:
We may also use "libre" word to avoid confusion of "open/free".

map-per replied:
I wouldn't use "libre" as too many people don't know what it means.

biodranik reviewed (commented):
Landscape screens look more complicated now. Would it be better to use something like:
```
title
icon | text
login block | register block
```
?

matheusgomesms commented on data/strings/strings.txt:
```diff
@@ -15968,50 +15968,98 @@
+    nb = OpenStreetMap.org (OSM) er et fellesskapsprosjekt for å bygge et gratis og åpent kart. Det er hovedkilden til kartdata i Organic Maps og fungerer på samme måte som Wikipedia. Du kan legge til eller redigere steder, og de blir tilgjengelige for millioner av brukere over hele verden. \nBli med i fellesskapet og bidra til å lage et bedre kart for alle!
+    nl = OpenStreetMap.org (OSM) is een gemeenschapsproject om een vrije en open kaart te bouwen. Het is de belangrijkste bron van kaartgegevens in Organic Maps en werkt ongeveer hetzelfde als Wikipedia. U kunt plaatsen toevoegen of bewerken en deze worden dan beschikbaar voor miljoenen gebruikers over de hele wereld.\nSluit u aan bij de gemeenschap en help mee om een betere kaart voor iedereen t ...
+    pl = OpenStreetMap.org (OSM) to projekt społecznościowy mający na celu stworzenie darmowej i otwartej mapy. Jest to główne źródło danych mapowych w Organic Maps i działa podobnie do Wikipedii. Można dodawać lub edytować miejsca, które stają się dostępne dla milionów użytkowników na całym świecie.\nDołącz do społeczności i pomóż stworzyć lepszą mapę dla wszystkich!
+    pt = O OpenStreetMap.org (OSM) é um projeto comunitário para construir um mapa livre e aberto. É a principal fonte de dados de mapas no Organic Maps e funciona de forma semelhante à Wikipedia. Pode adicionar ou editar locais e estes ficam disponíveis para milhões de utilizadores em todo o mundo.\nJunte-se à comunidade e ajude a criar um mapa melhor para todos!
+    pt-BR = O OpenStreetMap.org (OSM) é um projeto comunitário para criar um mapa gratuito e aberto. Ele é a principal fonte de dados de mapas no Organic Maps e funciona de forma semelhante à Wikipedia. Você pode adicionar ou editar lugares e eles ficam disponíveis para milhões de usuários em todo o mundo.\nJunte-se à comunidade e ajude a criar um mapa melhor para todos!
```
```suggestion
    pt = O OpenStreetMap.org (OSM) é um projeto comunitário aberto para construir um mapa gratuito. O OSM é a principal fonte de dados de mapas do Organic Maps e funciona de forma semelhante à Wikipedia. Pode adicionar ou editar locais e estes ficam disponíveis para milhões de utilizadores no mundo todo.\nJunte-se à comunidade e ajude a criar um mapa melhor para todos!
    pt-BR = O OpenStreetMap (OSM) é um projeto comunitário aberto para construir um mapa gratuito. O OSM é a principal fonte de dados de mapas do Organic Maps e funciona de forma semelhante à Wikipédia. Você pode adicionar ou editar lugares e eles ficam disponíveis para milhões de usuários no mundo todo.\nJunte-se à comunidade e ajude a criar um mapa melhor para todos!
```

map-per commented:
> Landscape screens look more complicated now. Would it be better to use something like:
> 
> ```
> title
> icon | text
> login block | register block
> ```
> 
> ?

IMO it the screen looks cleaner when title, icon, and text don't span over the whole with.

pastk commented:
To me both previous landscape layout and the new one are both fine. Happy either way :)

map-per commented:
On tablets it looks like this:
#### Before
![Bildschirmfoto vom 2024-10-26 16-38-45](https://github.com/user-attachments/assets/8023c586-2187-450c-80f7-356dbc0fcd8a)
#### Now
![Bildschirmfoto vom 2024-10-26 16-18-56](https://github.com/user-attachments/assets/e5bbd16e-5ac9-4b2b-86a5-6bd7916d9f36)

soshial commented on data/strings/strings.txt:
```diff
@@ -15968,50 +15968,98 @@
+    ja = OpenStreetMap.org(OSM)は、フリーでオープンな地図を構築するコミュニティ・プロジェクトです。OSMは、Organic Mapsの地図データの主要なソースであり、ウィキペディアと同じように機能します。あなたが場所を追加したり編集したりすると、世界中の何百万人ものユーザーがその場所を利用できるようになります！
+    ko = 오픈스트리트맵(OSM)은 무료 오픈스트리트맵을 구축하기 위한 커뮤니티 프로젝트입니다. 오가닉 맵의 지도 데이터의 주요 소스이며 위키백과와 유사하게 작동합니다. 장소를 추가하거나 편집하면 전 세계 수백만 명의 사용자가 사용할 수 있습니다.\n커뮤니티에 참여하여 모두를 위한 더 나은 지도를 만드는 데 도움을 주세요!
+    lt = OpenStreetMap.org (OSM) - tai bendruomenės projektas, kuriuo siekiama sukurti nemokamą ir atvirą žemėlapį. Tai pagrindinis žemėlapių duomenų šaltinis "Organiniuose žemėlapiuose" ir veikia panašiai kaip "Vikipedija". Galite pridėti ar redaguoti vietoves, ir jos tampa prieinamos milijonams naudotojų visame pasaulyje.\nPrisijunkite prie bendruomenės ir padėkite kurti geresnį žemėlapį visiem ...
+    lv = OpenStreetMap.org (OSM) ir kopienas projekts, kura mērķis ir izveidot bezmaksas un atvērtu karti. Tas ir galvenais karšu datu avots Organic Maps un darbojas līdzīgi kā Wikipedia. Jūs varat pievienot vai rediģēt vietas, un tās kļūst pieejamas miljoniem lietotāju visā pasaulē.\npievienojieties kopienai un palīdziet izveidot labāku karti visiem!
```
OpenStreetMap.org (OSM) ir kopienas projekts, kura mērķis ir izveidot atvērtajos datos balstītu karti, kura ir pieejama bez maksas. Tas ir Organic Maps galvenais karšu datu avots un darbojas līdzīgi kā Vikipēdija. Jūs varat pievienot vai rediģēt vietas, un tās kļūst pieejamas miljoniem lietotāju visā pasaulē.\nPievienojieties kopienai, lai palīdzētu izveidot labāku karti visiem!

NeatNit commented on data/strings/strings.txt:
```diff
@@ -15968,50 +15968,98 @@
+    fa = OpenStreetMap.org (OSM) یک پروژه اجتماعی برای ساخت یک نقشه رایگان و باز است. این منبع اصلی داده های نقشه در نقشه های ارگانیک است و مشابه ویکی پدیا کار می کند. میu200cتوانید مکانu200cها را اضافه یا ویرایش کنید تا در دسترس میلیونu200cها کاربر در سراسر جهان قرار گیرند.\nبه انجمن بپیوندید و به ایجاد نقشه بهتر برای همه کمک کنید!
+    fi = OpenStreetMap.org (OSM) on yhteisöprojekti, jonka tarkoituksena on rakentaa ilmainen ja avoin kartta. Se on Organic Mapsin tärkein karttatietojen lähde ja toimii samalla tavalla kuin Wikipedia. Voit lisätä tai muokata paikkoja, ja ne tulevat miljoonien käyttäjien saataville kaikkialla maailmassa.\nLiity yhteisöön ja auta tekemään parempi kartta kaikille!
+    fr = OpenStreetMap.org (OSM) est un projet communautaire visant à créer une carte libre et ouverte. C'est la principale source de données cartographiques d'Organic Maps et son fonctionnement est similaire à celui de Wikipédia. Vous pouvez ajouter ou modifier des lieux et ils deviennent accessibles à des millions d'utilisateurs dans le monde entier.\nRejoignez la communauté et aidez-nous à cré ...
+    he = OpenStreetMap.org (OSM) הוא פרויקט קהילתי לבניית מפה חופשית ופתוחה. זהו המקור העיקרי לנתוני מפות במפות אורגניות ופועל בדומה לוויקיפדיה. אתה יכול להוסיף או לערוך מקומות והם הופכים לזמינים למיליוני משתמשים בכל רחבי העולם.\nהצטרף לקהילה ועזור ליצור מפה טובה יותר לכולם!
```
```suggestion
    he = OpenStreetMap.org (OSM) הוא פרויקט קהילתי לבניית מפה חופשית ופתוחה. זהו המקור העיקרי לנתוני מפות ב-Organic Maps והוא פועל בדומה לוויקיפדיה. ניתן להוסיף או לערוך מקומות והם הופכים לזמינים למיליוני משתמשים בכל רחבי העולם.\nהצטרף לקהילה ועזור ליצור מפה טובה יותר לכולם!
```

Please take a screenshot of how Hebrew looks in the app, either before or after accepting my changes. It may be necessary to force the text direction in the code to be RTL instead of LTR.

If not done in code, it can be done in the text by adding a right-to-left mark U+200F to the start of the string. I don't know if adding Unicode characters with `\uXXXX` is possible. It's an invisible character so adding it literally could be trouble (for future editing).

```suggestion
    he = \u200FOpenStreetMap.org (OSM) הוא פרויקט קהילתי לבניית מפה חופשית ופתוחה. זהו המקור העיקרי לנתוני מפות ב-Organic Maps והוא פועל בדומה לוויקיפדיה. ניתן להוסיף או לערוך מקומות והם הופכים לזמינים למיליוני משתמשים בכל רחבי העולם.\nהצטרף לקהילה ועזור ליצור מפה טובה יותר לכולם!
```

map-per replied:
@NeatNit 
This is how Hebrew looks like now (with \u200F). Is the text displayed correctly?

![Screenshot_2024-11-01-15-24-40-767_app organicmaps debug](https://github.com/user-attachments/assets/7e8e9b7f-c806-41c4-9ca1-270611b0cc5f)

NeatNit replied:
Sorry for the late response. Yes, this is good.

jimcarst reviewed (commented):
nl ok

map-per commented:
Thank you for the reviews! I updated the language strings.
